src/gobapi/storage.py

```diff
@@ -218,6 +223,42 @@ def convert(entity):
     return convert
 
 
+def _add_resolve_attrs_to_columns(columns):
+    """Adds attributes to columns necessary to resolve model attributes from a view.
+
+    Looks for attributes of the form brk:sjt:heeft_bsn_voor, and will try to find the attribute heeft_bsn_voor in the
+    catalog brk and collection sjt.
+
+    Adds attribute, authority and public_name (heeft_bsn_voor in this case) to columns matching the pattern above.
+    """
+    resolve_attr_pattern = re.compile("^(\w+):(\w+):(\w+)$")
+    authorities = defaultdict(dict)
+
+    for column in columns:
+        match = re.match(resolve_attr_pattern, column.name)
+        if not match:
+            continue
+
+        catalog_abbreviation, collection_abbreviation, attribute_name = match.groups()
+
+        try:
+            catalog, collection = GOBModel().get_catalog_collection_from_abbr(catalog_abbreviation,
+                                                                              collection_abbreviation)
+            attribute = collection['attributes'][attribute_name]
+        except (NotInModelException, KeyError):
+            continue
+
+        authority = authorities[catalog_abbreviation].get(collection_abbreviation)
+
+        if not authority:
+            authority = Authority(catalog['name'], collection['name'])
+            authorities[catalog_abbreviation][collection_abbreviation] = authority
+
+        setattr(column, 'attribute', attribute)
+        setattr(column, 'authority', authority)
```

Comment:
Why not simply use setattr(column, 'authority',  Authority(catalog['name'], collection['name']))
There is no overhead involved by simply instantiating an Authority

Reply:
On second thougth, authority has two functions: authorize access and decrypt values.
The two functions are decoupled; the one function is filter_row and the other one is get_secured_value.
So when finally getting the value for an enhanced view GOB storage should 1. ask the authority if the request has access to the attribute and 2. decrypt the attribute value. Note that the authorisation scheme is a variable, the authority can in theory be instantiated with another authorisation scheme. The decryption is independent from the authorisation scheme (as this would not allow for changing the scheme).
Probably we need a more clever Authority to handle enhanced views, that check the scheme and decrypt the value in one method.

Reply:
Changed this as discussed. Moved authority logic to get_gob_value